features/telemetry/protocols/bmp/stats/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/stats/RouteInfoClient.java

```diff
@@ -0,0 +1,150 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.telemetry.protocols.bmp.adapter.stats;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import org.opennms.netmgt.telemetry.protocols.bmp.persistence.api.BmpRouteInfo;
+import org.opennms.netmgt.telemetry.protocols.bmp.persistence.api.BmpRouteInfoDao;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+public class RouteInfoClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RouteInfoClient.class);
+
+    private final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+            .setNameFormat("UpdateRouteInfo-%d")
+            .build();
+
+    private final ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor(
+            threadFactory);
+
+    private final String routeInfoDbPath;
+
+    private BmpRouteInfoDao bmpRouteInfoDao;
+
+
+    public RouteInfoClient(String routeInfoDbPath) {
+        this.routeInfoDbPath = routeInfoDbPath;
+    }
+
+    public void init() {
+        scheduledExecutorService.scheduleAtFixedRate(() -> updateRouteInfo(routeInfoDbPath), 1, 24, TimeUnit.HOURS);
+    }
+
+    public void destroy() {
+        scheduledExecutorService.shutdown();
+    }
+
+
+    private void updateRouteInfo(String folderName) {
+
+        if (Strings.isNullOrEmpty(folderName)) {
+            return;
+        }
+        List<RouteInfo> routeInfoList = new ArrayList<>();
+        try (Stream<Path> paths = Files.walk(Paths.get(folderName))) {
+
+            paths.filter(Files::isRegularFile).forEach(path -> {
+                List<RouteInfo> returned = parseEachFile(path);
+                routeInfoList.addAll(returned);
+            });
+        } catch (IOException e) {
+            LOG.error("Exception while walking through files in folder {}", folderName, e);
+        }
+        saveAndUpdateInDB(routeInfoList);
+    }
+
+    @VisibleForTesting
+    List<RouteInfo> parseEachFile(Path dbPath) {
+        try {
+            List<String> lines = Files.readAllLines(dbPath);
```

Comment:
Is it worth to make this streaming?

Reply:
Didn't get what you mean here? Please elaborate.

Reply:
Using `Files.lines(...)` gives you a stream of lines (allowing to process the file line by line) avoiding to read the whole file into memory before iterating over it. Looks like `RouteInfo.parseRouteInfo()` can have a `Stream<String>` or an `Iterator<String>` as parameter. Maybe this can be enhanced even more to also return a `Stream` from `RouteInfo.parseRouteInfo` if these objects don't have dependencies to each other while processing further down the line.

I have no clue how big these files usually are. Maybe they are just small an it's not worth the effort.

Reply:
Makes sense, fixed